pkg/operator/ceph/controller/spec.go

```diff
@@ -570,12 +570,18 @@ func GenerateLivenessProbeExecDaemon(daemonType, daemonID string) *v1.Probe {
 				//
 				// Example:
 				// env -i sh -c "ceph --admin-daemon /run/ceph/ceph-osd.0.asok status"
+				//
+				// Ceph gives pretty un-diagnostic error message when `ceph status` or `ceph mon_status` command fails.
+				// We'd better ignore this message and give users more simple one.
+				// The workaround below is bit dirty; but it works.
+				// ref: https://github.com/rook/rook/issues/9846
 				Command: []string{
 					"env",
 					"-i",
 					"sh",
 					"-c",
-					fmt.Sprintf("ceph --admin-daemon %s %s", confDaemon.buildSocketPath(), confDaemon.buildAdminSocketCommand()),
+					fmt.Sprintf("ceph --admin-daemon %s %s 2>/dev/null; if [ \\$? -ne 0 ]; then echo %s; exit 1; fi",
```

Comment:
I am concerned that this will destroy any error message including potentially valid ones. This also destroys the original return code. 

```suggestion
					fmt.Sprintf("ceph --admin-daemon %s %s; rc=\\$?; if [ \\$rc -ne 0 ]; then echo %s; exit \\$rc; fi",
```

Reply:
We could get a little more fancy with this if it seems helpful to put "ceph daemon health check failed" before any returned text from Ceph. That would be something like this:

```sh
```suggestion
					fmt.Sprintf(`outp="\\$(ceph --admin-daemon %s %s 2>&1); rc=\\$?; if [ \\$? -ne 0 ]; then echo %s; echo "\\$oupt"; exit \\$rc; fi`,
```

Reply:
Fair point, thanks. Thought it's better to quote Ceph's output so changed it to do so (and also, fix of shell script's bugs).